### PR TITLE
Xrc part02

### PR DESCRIPTION
**Pull request by wsu-cb**
Here are some more conversions of CBDesign dialogs from MFC to wx.

Note that these include some uses of `wxVListBox`.  In keeping with approximately the same layout as the original dialogs, none of the current replacements has enough height to demonstrate the behavior of `wxVListBox` with tiles of height > 255 pixels.  However, I did experiment with changing the height of the Create Marker dialog enough to verify that `wxVListBox` does correctly display tiles 350 pixels high when the `wxVListBox` is tall enough to show the entire tile.  However, the scrolling behavior of `wxVListBox` is not ideal when a tile exceeds the height of the window:  `wxVListBox` does not support scrolling of partial lines; the top of the topmost visible item must always be at the top of the `wxVListBox`, so it is still impossible to see the bottom of a tile if the tile exceeds the `wxVListBox` height.

### CODE REVIEW

**Comment by wsu-cb** [comment]
Closed/reopened PR to force checks to run again because, for reasons I don't understand, the checks were interrupted.

File: GShr/LBoxMark.cpp [hunk 1]
```diff
@@ -254,3 +260,197 @@ BOOL CMarkListBox::OnDragSetup(DragInfo& pDI) const
...
+{
+    if (m_eTrayViz == mtrayVizNormal)
+    {
+        if (m_pDoc == NULL)
+            return;
+
+        CMarkManager* pMMgr = m_pDoc->GetMarkManager();
+        if (pMMgr == NULL)
+        {
+TRACE("handled null pMMgr\n");
```
Comment (DLLarson) [inline GShr/LBoxMark.cpp:416]: Indent TRACE.

**Comment by DLLarson** [comment]
>However, the scrolling behavior of wxVListBox is not ideal when a tile exceeds the height of the window: wxVListBox does not support scrolling of partial lines; the top of the topmost visible item must always be at the top of the wxVListBox, so it is still impossible to see the bottom of a tile if the tile exceeds the wxVListBox height.

What did the Windows do when it couldn't show the entire item of the 250-ish height limit?

-Dale

**Review by DLLarson — APPROVED** [review]
I'm done. Not much to comment on. Except for the wxWidgets virtual list box most is just one-for-one switch overs from MFC to wxWidgets.

Which method of merge would you prefer?

-Dale

File: GShr/LBoxGrfx.h [hunk 2]
```diff
@@ -428,5 +429,352 @@ class CGrafixListBoxData : public BASE_WND
...
+class CGrafixListBoxWx : public CB::VListBoxHScroll
+{
+    // Construction
+public:
+    CGrafixListBoxWx();
+    ~CGrafixListBoxWx() override;
+
+    // Attributes
+public:
+#if 0
```
[the remark this answers is not in the document]
Comment (wsu-cb) [reply GShr/LBoxGrfx.h:448]: If I left any in that aren't needed, that was a mistake.  However, I have definitely used `#if 0` for things that aren't needed YET, but which I expect will be needed eventually.

**Comment by wsu-cb** [comment]
> > However, the scrolling behavior of wxVListBox is not ideal when a tile exceeds the height of the window: wxVListBox does not support scrolling of partial lines; the top of the topmost visible item must always be at the top of the wxVListBox, so it is still impossible to see the bottom of a tile if the tile exceeds the wxVListBox height.
> 
> What did the Windows do when it couldn't show the entire item of the 250-ish height limit?
> 

I didn't remember, so I had to go recompile that version.  It turns out the MFC version behaves the same way.

File: GShr/LBoxMark.cpp (hunk 1 above)
Comment (wsu-cb) [reply GShr/LBoxMark.cpp:416]: Actually, if I do anything, I will just remove this line.  Do you want me to do that before merging the PR?

**Comment by wsu-cb** [comment]
> I'm done. Not much to comment on. Except for the wxWidgets virtual list box most is just one-for-one switch overs from MFC to wxWidgets.
> 
> Which method of merge would you prefer?
> 

I prefer keeping the individual commits, but if you prefer to squash them, it won't bother me very much.

**Comment by DLLarson** [comment]
>I prefer keeping the individual commits, but if you prefer to squash them, it won't bother me very much.

Your wish is my command...after the final push is done that is.

File: GShr/LBoxMark.cpp (hunk 1 above)
Comment (wsu-cb) [reply GShr/LBoxMark.cpp]: done

**Pull request merged**